src/Microsoft.Deployment.DotNet.Releases/src/ReleaseVersion.cs

```diff
@@ -568,5 +559,281 @@ internal static int CompareNumericIdentifier(string a, string b)
+        /// <returns><see langword="true"/> if the string only contains digits; <see langword="false"> otherwise.</see></returns>
+        internal static bool IsDigits(string input)
+        {
+            if (string.IsNullOrWhiteSpace(input))
+            {
+                return false;
+            }
+
+            for (int i = 0; i < input.Length; ++i)
+            {
+                if (!char.IsDigit(input[i]))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Determines whether a string is a valid build identifier.
+        /// </summary>
+        /// <param name="input">The string to check.</param>
+        /// <returns><see langword="true"/> if the string is a build identifier; <see langword="false"> otherwise.</see></returns>
+        internal static bool IsBuildIdentifier(string input)
+        {
+            return IsAlphaNumericIdentifier(input) || IsDigits(input);
+        }
+
+        /// <summary>
+        /// Determines whether a string is a valid alphanumeric identifier.
+        /// </summary>
+        /// <param name="input">The string to check.</param>
+        /// <returns><see langword="true"/> if the string is an alphanumeric identifier; <see langword="false"> otherwise.</see></returns>
+        internal static bool IsAlphaNumericIdentifier(string input)
+        {
+            if (string.IsNullOrWhiteSpace(input))
+            {
+                return false;
+            }
+
+            for (int i = 0; i < input.Length; i++)
+            {
+                if (!(char.IsLetterOrDigit(input[i]) || input[i] == '-'))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Determines whether a string is a valid prelease identifier.
+        /// </summary>
+        /// <param name="input">The string to check.</param>
+        /// <returns><see langword="true"/> if the string is a prerelease identifier; <see langword="false"> otherwise.</see></returns>
+        internal static bool IsPrereleaseIdentifier(string input)
+        {
+            return IsNumericIdentifier(input) || IsAlphaNumericIdentifier(input);
+        }
+
+
+        /// <summary>
+        /// Tries to convert the string representation of a version to an equivalent <see cref="ReleaseVersion"/> object,
+        /// and returns a value indicating whether the conversion succeeded. 
+        /// </summary>
+        /// <param name="input">A string that contains a version to convert.</param>
+        /// <param name="version">Contains the <see cref="ReleaseVersion"/> equivalent of the value in <paramref name="input"/> if
+        /// successful; otherwise <see langword="null"/>.</param>
+        /// <param name="canThrow">A boolean indicating whether an exception should be raised if the conversion failed.</param>
```

Comment:
Consider renaming to throwOnFailure.  canThrow indicates that an exception may not be thrown if the version cannot be parsed.